python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -76,6 +80,13 @@ def groups(self):
         """
         group_names, offsets, _, grouped_values = self._grouped()
         grouped_index = grouped_values.index
+
+        if len(group_names) > self._MAX_GROUPS_BEFORE_WARN:
+            warnings.warn(
+                f"GroupBy.groups() expected to be slow"
+                f" for large number of groups ({len(group_names)})"
```

Comment:
We should make this more clear what the number is: because right now it would output:
```
GroupBy.groups() expected to be slow for large number of groups 1000
```

Maybe reword to:
```
GroupBy.groups() performance scales poorly with a large number of groups: got 1000 groups
```?

Reply:
Yeah I definitely like warning message better. Changed it to that